src/main.rs

```diff
@@ -45,7 +46,40 @@ async fn main() -> std::result::Result<(), Box<dyn std::error::Error>> {
     let url = Url::new(url, args.default_scheme);
     let host = url.host().unwrap();
     let method = method.unwrap_or(Method::from(&body)).into();
-    let auth = Auth::new(args.auth, args.auth_type, &host);
+    let mut auth = Auth::new(args.auth, args.auth_type, &host);
+
+    // load previous session if present
+    let arg_session = args.session.clone();
+    let previous_session = match args.session {
+        None => None,
+        Some(identifier) => {
+            match Session::load(&identifier, &host) {
+                Err(why) => panic!("couldn't load session {}: {}", &identifier, why),
+                Ok(result) => result,
+            }
+        },
+    };
+    let session_for_merge = previous_session.clone();
+    // Use auth from previous session if no auth present
+    match (auth.is_none(), previous_session) {
```

Comment:
If being rid of the empty match arm is desired, I believe something like the following could be used.
```rust
if let (true, Some(p)) = (auth.is_none(), previous_session) {
    auth = p.auth;
}
```

Reply:
Sure, thanks.